internal/driver/commands.go

```diff
@@ -360,70 +353,52 @@ var kcachegrind = []string{"kcachegrind"}
 // the terminal screen. This is used to avoid dumping binary data on
 // the screen.
 func awayFromTTY(format string) PostProcessor {
-	return func(input []byte, output io.Writer, ui plugin.UI) error {
+	return func(input io.Reader, output io.Writer, ui plugin.UI) error {
 		if output == os.Stdout && (ui.IsTerminal() || interactiveMode) {
 			tempFile, err := newTempFile("", "profile", "."+format)
 			if err != nil {
 				return err
 			}
 			ui.PrintErr("Generating report in ", tempFile.Name())
-			_, err = fmt.Fprint(tempFile, string(input))
-			return err
+			output = tempFile
 		}
-		_, err := fmt.Fprint(output, string(input))
+		_, err := io.Copy(output, input)
 		return err
 	}
 }
 
 func invokeDot(format string) PostProcessor {
-	divert := awayFromTTY(format)
-	return func(input []byte, output io.Writer, ui plugin.UI) error {
+	return func(input io.Reader, output io.Writer, ui plugin.UI) error {
 		cmd := exec.Command("dot", "-T"+format)
-		var buf bytes.Buffer
-		cmd.Stdin, cmd.Stdout, cmd.Stderr = bytes.NewBuffer(input), &buf, os.Stderr
+		cmd.Stdin, cmd.Stdout, cmd.Stderr = input, output, os.Stderr
 		if err := cmd.Run(); err != nil {
 			return fmt.Errorf("Failed to execute dot. Is Graphviz installed? Error: %v", err)
 		}
-		return divert(buf.Bytes(), output, ui)
+		return nil
 	}
 }
 
-func saveSVGToFile() PostProcessor {
+func invokeDotSVG() PostProcessor {
```

Comment:
Please add a comment explaining why this is different from invokeDot("svg"): mainly that it also massages the SVG to add the svgpan functionality.

Also maybe rename it, something like massageDotSVG, or invokeDotSvgpan ?